src/features/ci-cd/hooks/service-installs.ts

```diff
@@ -155,27 +231,68 @@ hooks.addPureHook('POST', 'resin', 'device', {
 });
 
 hooks.addPureHook('PATCH', 'resin', 'device', {
-	POSTRUN: async (args) => {
-		const affectedIds = args.request.affectedIds!;
+	PRERUN: async (args) => {
+		const affectedDeviceIds = args.request.affectedIds!;
 
-		// We need to delete all service_install resources for the current device and
+		// We need to delete all service_install resources for the current device / app and
 		// create new ones for the new application (if the device is moving application)
-		if (
-			args.request.values.belongs_to__application != null &&
-			affectedIds.length !== 0
-		) {
-			await args.api.delete({
-				resource: 'service_install',
+		const newAppId = args.request.values.belongs_to__application;
+		if (newAppId && affectedDeviceIds.length !== 0) {
+			const createHandler = async (
+				api: sbvrUtils.PinejsClient,
+				deviceIds: number[],
+				serviceIds: number[],
+			) => {
+				await Promise.all(
+					deviceIds.map(async (deviceId) => {
+						await Promise.all(
+							serviceIds.map(async (serviceId) => {
+								// Create a service_install for this pair of service and device
+								await api.post({
+									resource: 'service_install',
+									body: {
+										device: deviceId,
+										installs__service: serviceId,
+									},
+									options: { returnResource: false },
+								});
+							}),
+						);
+					}),
+				);
+			};
```

Comment:
This can now go back to how it was before right? Other than keeping the optimization of course

Reply:
The optimization comes from using a separate handler in "moving to a different app" case.  If I put everything back, the optimization goes away.  

Specifically the optimization is that when we delete all service installs, we don't need to get through the process of getting existing ones and creating the difference.  The difference in behavior is because we are using a different handler (since the rest of the process is the same).   The special case handler creates _all_ the service installs for the app instead of doing the comparison.